js-core-3/week-2/lesson.md

```diff
@@ -1,305 +1,302 @@
-> if gender is 'female' or not, and use it in sayGreeting. Let's try and make
-> the code as compact as possible together using ES6 features.
+**scripts.js**
 
+```javascript
+// This function should retrieve the JSON from the `countryURL` and then call onCountryDataReceived() with the JSON
+function getData(countryURL) {}
 
-## Default parameters 
+function onCountryDataReceived(country) {
+  addCountryName(country);
+  addCountryCapital(country);
+  addNameInOtherLanguages(country);
+}
 
-ES6 allows us to declare defaults for function arguments. The default value is 
-used when the argument/parameter is either missing or `undefined`. 
+// This function should take the JSON for the country and put a H1 tag on the screen containing its name
+function addCountryName(countryData) {}
 
-This function returns the sum of three numbers. Let's assume we want to use the 
-same function with only two arguments: 
+// This function should take the JSON for the country and put a H2 tag on the screen containing its capital city
+function addCountryCapital(countryData) {}
 
-```js
-// without default parameter
-const sum = (a, b, c) => {
-  c = c || 0;
-  return a + b + c;
+// This function should take the JSON for the country and put UL and LI tags on the screen with the countries name translated into other languages
+function addNameInOtherLanguages(countryData) {}
+
+function getContentDiv() {
+  return document.querySelector("#content");
 }
 
-console.log(sum(1, 3, 4)) // 8
-console.log(sum(2, 5)) // NaN
+function onLoad() {
+  getData(
+    "https://restcountries.eu/rest/v2/name/Great%20Britain?fullText=true"
+  );
 
-// with default parameter 
-const sum = (a, b, c = 0) => {
-  return a + b + c;
-}
-console.log(sum(2, 5)) // 7
-```
+  /** Remove this line when you have completed the task
+
+  getData("https://restcountries.eu/rest/v2/name/France?fullText=true");
+
+  getData("https://restcountries.eu/rest/v2/name/Germany?fullText=true");
+
+  getData("https://restcountries.eu/rest/v2/name/Spain?fullText=true");
 
+  getData("https://restcountries.eu/rest/v2/name/Portugal?fullText=true");
 
-## Destructuring
+  getData("https://restcountries.eu/rest/v2/name/Hungary?fullText=true");
 
-In ES6 we can extract data from objects or arrays using destructuring. 
+  getData("https://restcountries.eu/rest/v2/name/Russia?fullText=true");
 
-```js
-// before 
-var chicken = {
-  name: 'Maggie', 
-  age: 2
+  */ // Remove this line when you have completed the task
 }
 
-var name = chicken.name;
-var age = chicken.age;
+window.onload = onLoad;
+```
 
-var numbers = [1, 2];
+##### Finished Example
 
-var firstNumber = numbers[0];
-var secondNumber = numbers[1];
+You can find the finished example of this website [here](js-core-3\week-1\completed_country_website).
 
+##### Check-in
 
-// in ES6
-const chicken = {
-  name: 'Maggie', 
-  age: 2
-}
+**Question (5 mins):**
 
-const { name, age } = chicken;
+    Which of the following statements about Promises is false?
```

Comment:
I don't think this syllabus has covered enough about promises for the students to be able to answer these questions... The lesson may, unscripted, but...